controllers/apps/transformer_component_workload_upgrade.go

```diff
@@ -120,6 +96,30 @@ func (t *componentWorkloadUpgradeTransformer) Transform(ctx graph.TransformConte
 		}
 	}
 
+	// remove the StatefulSet object if found
+	sts := &appsv1.StatefulSet{}
+	if err := graphCli.Get(transCtx.Context, client.ObjectKeyFromObject(comp), sts); err == nil {
+		legacyFound = true
+		parent = graphCli.Do(dag, nil, sts, model.ActionDeletePtr(), parent, model.WithPropagationPolicy(client.PropagationPolicy(metav1.DeletePropagationOrphan)))
+	} else if !apierrors.IsNotFound(err) {
+		return err
+	}
+
```

Comment:
I think the order of deleting the StatefulSet and deleting the RSM doesn't matter. Deleting the StatefulSet first and then the RSM, or deleting the RSM first and then the StatefulSet, both are fine, right?

Reply:
that's correct.
however the handling order of the StatefulSet and its Pods does matter. should delete the StatefulSet first(with `PropagationPolicy=Orphan`), then we can manipulate all the `metadata` part and `spec` part of the Pods safely.